tests/test_router_quote.py

```diff
@@ -0,0 +1,244 @@
+import time
+import pytest
+
+import conftest
+from sqs_service import *
+from quote_response import *
+from rand_util import *
+from e2e_math import *
+from decimal import *
+from constants import *
+from util import *
+
+SQS_STAGE = "https://sqs.stage.osmosis.zone"
+
+ROUTES_URL = "/router/quote"
+
+QUOTE_NUM_TOP_LIQUIDITY_DENOMS = 20
+
+# Minimum liquidity of each token in the transmuter pool for the test to be valid.
+# Tranmsuter USDC pool frequently gets imbalanced since people move from
+# USDC.axl to USDC.noble. Without this limit, the test would fail frequently.
+TRANSMUTER_MIN_TOKEN_LIQ_USD = 10000
+
+# Arbitrary choice based on performance at the time of test writing
+expected_latency_upper_bound_ms = 1000
+
+# Test suite for the /router/quote endpoint
+class TestQuote:
+    @pytest.mark.parametrize("coin_obj", construct_token_in_combos(conftest.choose_tokens_liq_range(QUOTE_NUM_TOP_LIQUIDITY_DENOMS), USDC_PRECISION - 1, USDC_PRECISION + 4), ids=id_from_coin)
+    def test_usdc_in_high_liq_out(self, environment_url, coin_obj):
+        """
+        This test case validates quotes betwen USDC in and NUM_TOP_LIQUIDITY_DENOMS.
+        The amounts are constructed to be seeded random values between 10^USDC_PRECISION-1 and 10 ^(USDC_PRECISION + 4)
+
+        This allows us to validate that we can continue to quote at reasonable USDC values for all majore token pairs without errors.
+
+        Note: the reason we use Decimal in this test is because floats truncate in some edge cases, leading
+        to flakiness.
+        """
+        # This is the max error tolerance of 7% that we allow.
+        # Arbitrarily hand-picked to avoid flakiness.
+        error_tolerance = 0.07
```

Comment:
How about `error_tolerance` move to constants.py ?

Reply:
This is by design. Each test benefits from having a custom error tolerance. They don't have a universal one shared across. Note that some of them already have different values.

We will continue tweaking the one for each based on the needs basis.

Reply:
OK. Let's keep it like that and we will revisit it as needed.